### ISSUE TITLE
Fix member function signatures

### ISSUE THREAD

**Issue by jklmnn** [Componolit/capdpa#38]
All member functions in Ada require a first argument that passes the instance.
E.g. `void C::f()` becomes `procedure f (This : access C)`.

**Comment by senier** [comment]
The first intuition is to add the `this` pointer to the IR. This, however, is problematic when it comes to name mangling. Say we have this class:

```C++
class Cls                                                                                                                                                                                      
{
   private:
      int a;

   public:
      Cls();
      int foo (int arg);
};
```

The method `foo` is exported as symbol `_ZN3Cls3fooEi`, i.e. with exactly one integer parameter (the last `i` - return values are not mangled for methods). When we add the `this` pointer to the IR, the mangling produces `_ZN3Cls3fooERS_i` which is a reference to the class (`S_` refers to `Cls`) and an integer parameter `i`. This is correct for a normal parameter, but not what we need here.

I see two options here:

1. Leave the `this` pointer and perform special handling in the mangling code. (Depending on what? The name of the parameter? A special flag?)
2. Remove the `this` pointer from the IR and only add it during program generation. As the `Class` record is always in the same package and local to the methods this could simple be static.

@jklmnn: Opinions?

**Comment by senier** [comment]
Option 2 seems to work pretty well. I just generated the first automatic binding of the above class using Cappulada. A `<project>.ads` file with the required types had to be created manually. I'll open a ticket on that shortly.

**Comment by jklmnn** [comment]
Setting the this argument in the Ada generation part is indeed the better option since it does not add information to the IR which was not present directly in the AST.
It is furthermore far easier to implement.
Yet we have to check for static functions which do not have a this argument (do they have something else or nothing?).

**Comment by senier** [comment]
We did not handle static members at all so far. Fixed in 5bed6aa.
